src/Data/Filterable.purs

```diff
@@ -67,6 +72,22 @@ partitionDefault p xs =
   let o = partitionMap (eitherBool p) xs
   in {no: o.left, yes: o.right}
 
+-- | A default implementation of `partition` using `filter`. Note that this is
+-- | almost certainly going to be suboptimal compared to direct implementations.
+partitionDefaultFilter :: forall f a. Filterable f =>
+  (a -> Boolean) -> f a -> { no :: f a, yes :: f a }
+partitionDefaultFilter p xs = { yes: filter p xs, no: filter (not p) xs }
+
+-- | A default implementation of `partition` using `filterMap`. Note that this
+-- | is almost certainly going to be suboptimal compared to direct
+-- | implementations.
+partitionDefaultFilterMap :: forall f a. Filterable f =>
```

Comment:
Could you move this somewhere under `maybeBool`?  I try not to use declarations until after they have been introduced.